airflow-core/src/airflow/api_fastapi/common/parameters.py

```diff
@@ -815,6 +817,93 @@ def _transform_dag_run_types(types: list[str] | None) -> list[DagRunType | None]
     _SearchParam, Depends(search_param_factory(DagRun.triggering_user_name, "triggering_user"))
 ]
 
+
+class _DagRunConsumingAssetFilter(BaseParam[str | None]):
+    """
+    Filter DagRuns by an asset they consumed (triggered by).
+
+    Matches by asset name or URI substring on related consumed_asset_events.
+    """
+
+    def __init__(self) -> None:
+        super().__init__(skip_none=True)
+
+    @classmethod
+    def depends(
+        cls,
+        consuming_asset: str | None = Query(
+            None, description="Filter DagRuns that consumed an asset (match by asset name or URI)"
+        ),
+    ) -> _DagRunConsumingAssetFilter:
+        return cls().set_value(consuming_asset)
+
+    def to_orm(self, select: Select) -> Select:  # type: ignore[name-defined]
+        if not self.value:
+            return select
+        asset_match = or_(AssetModel.name.ilike(f"%{self.value}%"), AssetModel.uri.ilike(f"%{self.value}%"))
+        exists_clause = (
+            sql_select(1)
+            .select_from(dagrun_asset_event_table)
+            .join(AssetEvent, AssetEvent.id == dagrun_asset_event_table.c.event_id)
+            .join(AssetEvent.asset)
+            .where(
+                and_(
+                    dagrun_asset_event_table.c.dag_run_id == DagRun.id,
+                    asset_match,
+                )
+            )
+            .exists()
+        )
+        return select.where(exists_clause)
+
+
+class _DagRunProducingAssetFilter(BaseParam[str | None]):
```

Comment:
Similarly to other 'search' operating on `ilike` `Pattern` suffix should be in the name, take a look at other search params in the same file.